presto-matching/src/main/java/com/facebook/presto/matching/v2/Capture.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.matching.v2;
+
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class Capture<T>
+{
+    private static final AtomicInteger sequenceCounter = new AtomicInteger();
+
+    private final int sequenceNumber;
+
+    public static <T> Capture<T> newCapture()
+    {
+        return new Capture<>();
+    }
+
+    private Capture()
+    {
+        this.sequenceNumber = sequenceCounter.incrementAndGet();
```

Comment:
Add string optional param to newCapture, suffix with sequence number. Sequence number is needed to not get lost when comparing 'Pattern.toString()' with the objects in debuger.